lib/client-base.js

```diff
@@ -474,7 +484,11 @@ function connect (client) {
 
       const err = socket[kError]
 
-      parser.destroy(err)
+      if (!client.running && !isConnectionError(err)) {
```

Comment:
client.running is in order to only apply this if no request is currently running which might have caused some other error, e.g. aborted.

Reply:
We could also turn this around and error on errors we know are unrecoverable... not sure which approach is best... unrecoverable might be a long list...